### DISCUSSION TITLE
Best way to enable Docker

### DISCUSSION THREAD

**Discussion by aschilling** [community/community#38264] [Codespaces] [answered]
Hi everybody,

I am currently using Codespaces with the Baseimage for Java development (mcr.microsoft.com/vscode/devcontainers/java:17) for my Dockerfile what is the best way for this image to install the Docker engine and Docker client within my Dockerfile? Do you have any recommendations or samples?

Best
Andy

**Comment by aschilling** [comment, marked as answer]
I found a great way myself to enable Docker. In particular, you can use "features" in the devcontainer.json to enable Docker (see here https://github.com/devcontainers/features)

**Comment by michielbdejong** [comment]
Please be aware that Codespaces is still in Beta, and using Docker inside it is apparently still buggy, e.g. this issue which we ran into: https://github.com/orgs/community/discussions/38878
